components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -1394,6 +1398,57 @@ public List<String> getOrganizationsMetaAttributes(boolean recursive, Integer li
         return organizationMetaAttributes;
     }
 
+    @Override
+    public Map<String, BasicOrganization> getBasicOrganizationDetailsByOrgIDs(List<String> orgIds)
+            throws OrganizationManagementException {
+
+        String placeholders = orgIds.stream()
+                .map(id -> "?")
+                .collect(Collectors.joining(", "));
+
+        String sql = String.format(GET_BASIC_ORG_DETAILS_BY_ORG_IDS, placeholders);
+        NamedJdbcTemplate namedJdbcTemplate = Utils.getNewTemplate();
+
+        try {
+            Map<String, BasicOrganization> basicOrganizationDetailsMap = new HashMap<>();
+            List<String> invalidOrgIds = new ArrayList<>();
+
+            namedJdbcTemplate.executeQuery(sql, (resultSet, rowNumber) -> {
+                        String orgId = resultSet.getString(VIEW_ID_COLUMN);
+                        String orgName = resultSet.getString(VIEW_NAME_COLUMN);
+                        String status = resultSet.getString(VIEW_STATUS_COLUMN);
+                        String created = resultSet.getString(VIEW_CREATED_TIME_COLUMN);
+                        String organizationHandle = resultSet.getString(VIEW_TENANT_DOMAIN_COLUMN);
```

Comment:
Let's avoid unnecessary variable creation except for orgName and orgId, which is used in the if condition